rskj-core/src/main/java/co/rsk/config/NodeCliOptions.java

```diff
@@ -41,6 +52,62 @@ public Config withConfig(Config config, String configValue) {
             return config.withValue(SystemProperties.PROPERTY_BASE_PATH, ConfigValueFactory.fromAnyRef(configValue));
         }
     },
+    SYNC_MODE("sync-mode", true) {
+        @Override
+        public Config withConfig(Config config, String configValue) {
+            if(configValue.equalsIgnoreCase("full") || configValue.equalsIgnoreCase("snap")){
+                return config.withValue(RskSystemProperties.PROPERTY_SYNC_MODE, ConfigValueFactory.fromAnyRef(configValue));
+            }else {
+                throw new IllegalArgumentException("Invalid sync mode: " + configValue + ". The valid options are <full> or <snap>.");
+            }
+        }
+    },
+    SNAP_NODES("snap-nodes", true) {
+        @Override
+        public Config withConfig(Config config, String configValue) {
+            try {
+                List<ConfigObject> snapConfigObjects = Arrays.stream(configValue.substring(1, configValue.length() - 1).split(","))
+                        .map(String::trim)
+                        .filter(s -> !s.isEmpty())
+                        .map(this::createConfigObjectFromNode)
+                        .collect(Collectors.toList());
+
+                if(!snapConfigObjects.isEmpty()) {
+                    ConfigValue snapConfigValue = ConfigValueFactory.fromIterable(snapConfigObjects);
+                    return config.withValue(RskSystemProperties.PROPERTY_SNAP_NODES, snapConfigValue);
+                }
+                return config;
+            } catch (Exception e) {
+                e.printStackTrace();
+                return config;
+            }
+        }
+        private ConfigObject createConfigObjectFromNode(String snapNode) {
+            try {
+                URI snapURI = new URI(snapNode);
+                validateSnapNodeURI(snapURI);
+
+                String id = snapURI.getUserInfo();
+                String host = snapURI.getHost();
+                int port = snapURI.getPort();
+
+                Map<String, ConfigValue> map = new HashMap<>();
+                map.put("port", ConfigValueFactory.fromAnyRef(port));
+                map.put("ip", ConfigValueFactory.fromAnyRef(host));
+                map.put("nodeId", ConfigValueFactory.fromAnyRef(id));
+
+                return ConfigFactory.parseMap(map).root();
+            } catch (URISyntaxException e) {
+                throw new IllegalArgumentException("Invalid URL format: " + snapNode, e);
```

Comment:
```suggestion
                throw new IllegalArgumentException("Invalid URL format for snap node: " + snapNode, e);
```

Just to let it clear straight away from where come from this errorMessage.